pkg/service/backup/restore_worker_data.go

```diff
@@ -435,7 +435,7 @@ func (w *restoreWorker) newRunProgress(ctx context.Context, run *RestoreRun, tar
 		return nil, errors.Wrap(err, "validate free disk space")
 	}
 
-	takenIDs := w.chooseIDsForBatch(ctx, h.Shards, target.BatchSize)
+	takenIDs := w.chooseIDsForBatch(ctx, target.BatchSize)
```

Comment:
Was this change consulted with someone with more understanding on the importance of shard cnt in the process of load and stream?

Reply:
This change decreases the batch size actually.
`h.Shards` were used as a multiplier for a default batch size.
But actually you're right let me check it with scylla core team.

Reply:
> The number of shards on the node where you perform load and stream do not have impact on how we process the data.

Does this mean that the number of shards does not have significant influence on load and stream speed?
I'm wondering whether load and stream most expensive operation is the actual file transfer between the nodes, or maybe the process of deciding, to which node should the files be sent (I'm considering the `primary_replica_only` option). If the second case is more expensive, does the number of shards influence its speed?

Reply:
^^ @asias please take a look on last Michal's question.

Reply:
A node with 32 shards is definitely faster than a node with 2 shards because it has 16x more cpu powers.  But load and stream on 2 shards node is as efficient as on 32 shards. I guess you want to check if the later is true. 

Load and stream does not send files over network. It reads data (mutation fragments) from sstables and figure out which nodes own the data and send the mutation fragments to the owning node. 

The most expensive operations are the actual reading from disk and send over network. The deciding part is relatively cheap.

Reply:
@Michal-Leszczynski this comment is very relevant here as well https://github.com/scylladb/scylla-manager/pull/3300/files#r1150523568

Reply:
So maybe we want to replace shard count in work distribution process with file size? Well, not exactly files size as (for my understanding) it's safer to upload all SSTables with the same ID at once, so rather the size of the SSTable bundle? 

On the other hand, removing shard count and setting batch size to 1 would also force hosts to divide bundles in a more fair way. Right now hosts just take the bundles from bundle pool, restore them and then check if there are any more bundles left to restore, so if we treat all nodes equally, this should get a good work distribution.

So I agree with removing shard count form the restore procedure.

Reply:
> On the other hand, removing shard count and setting batch size to 1 would also force hosts to divide bundles in a more fair way. Right now hosts just take the bundles from bundle pool, restore them and then check if there are any more bundles left to restore, so if we treat all nodes equally, this should get a good work distribution.

I think deciding if we should set batch_size to 1 instead of 2 is something what we can fine tune later (with the next minor releases).
Let's stay with default batch_size = 2, and let's remove this shards_count dependency at all.

Reply:
So it would be nice to remove it from `restoreHosts` an `initHosts` as well!